kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtComponentServiceImpl.java

```diff
@@ -244,12 +250,20 @@ public GwtConfigComponent findWireComponentConfigurationFromPid(GwtXSRFToken xsr
                 if (conf != null) {
                     conf.getConfigurationProperties().put(ConfigurationAdmin.SERVICE_FACTORYPID, factoryPid);
                 }
-                if ((conf != null) && (conf.getDefinition() == null)) {
+                if (conf != null && conf.getDefinition() == null) {
                     String temporaryName = String.valueOf(System.nanoTime());
                     cs.createFactoryConfiguration(factoryPid, temporaryName, extraProps, false);
                     try {
-                        // wait for the services to be up
-                        TimeUnit.MILLISECONDS.sleep(500);
+
```

Comment:
If you check `ServiceTracker` closely, it optimizes itself and checks if it can be run in separate thread. But in this scenario, it will not spawn new thread to wait for the service as it simply calls `Thread.wait` internally in the service instance to be tracked. Go for it 👍 . Just make the timeout local to the method.